misc/peer-id-generator/src/main.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+extern crate libp2p_core;
+extern crate libp2p_secio;
+extern crate num_cpus;
+extern crate rand;
+
+use libp2p_core::PeerId;
+use libp2p_secio::SecioKeyPair;
+use std::{env, str, thread, time::Duration};
+
+fn main() {
+    let prefix = match env::args().nth(1) {
+        Some(prefix) => prefix,
+        None => {
+            println!("Usage: {} <prefix>", env::args().nth(0).unwrap());
```

Comment:
"Usage {} <prefix>, where prefix is a sequence of letters/numbers from the Base58 alfabet, starting with one of `NPQRSTUVWXYZ`"